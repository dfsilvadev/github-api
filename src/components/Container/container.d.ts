import { ReactNode } from "react";

export interface ContainerProps {
  children: ReactNode;
  size?: "fixed" | "fluid";
}
