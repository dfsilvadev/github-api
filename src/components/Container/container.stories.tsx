import { Meta, Story } from "@storybook/react/types-6-0";
import { ComponentProps } from "react";

import Container from ".";

const children = (
  <div
    style={{
      border: "1px solid #fff"
    }}
  >
    Container
  </div>
);

export default {
  title: "Layout/Container",
  component: Container,
  argTypes: {
    children: {
      name: "Container children",
      description: "O children define o conteúdo do button."
    },
    size: {
      name: "Container size",
      description: "Referente ao tamanho do container.",
      options: ["fixed", "fluid"],
      control: { type: "select" },
      table: {
        defaultValue: {
          summary: "fluid"
        },
        type: {
          summary: "Mostrar opções de tamanho para o container.",
          detail: "fixed, fluid"
        }
      }
    }
  },
  args: {
    children,
    size: "fluid"
  },
  parameters: {
    layout: "fullscreen"
  }
} as Meta;

export const Default: Story<ComponentProps<typeof Container>> = ({
  ...props
}) => (
  <div
    style={{
      width: "100%"
    }}
  >
    <Container {...props} />
  </div>
);
