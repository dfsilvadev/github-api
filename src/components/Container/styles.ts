import styled, { DefaultTheme, css } from "styled-components";

import { ContainerProps } from "./container";

const wrapperModifiers = {
  fixed: (theme: DefaultTheme) => css`
    max-width: ${theme.grid.container};
  `,
  fluid: () => css`
    width: 100%;
  `
};

export const Wrapper = styled.div<Pick<ContainerProps, "size">>`
  ${({ theme, size }) => css`
    height: 100%;
    margin: 0 auto;
    padding: ${theme.spacings.large} ${theme.spacings.xsmall};

    ${!!size && wrapperModifiers[size](theme)}
  `}
`;
