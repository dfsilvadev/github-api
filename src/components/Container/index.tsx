import * as S from "./styles";

import { ContainerProps } from "./container";

const Container = ({ children, size = "fluid" }: ContainerProps) => {
  return <S.Wrapper size={size}>{children}</S.Wrapper>;
};

export default Container;
