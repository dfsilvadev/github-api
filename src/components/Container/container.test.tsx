import { renderWithTheme } from "../../utils/tests/renderWithTheme";

import Container from ".";
import { screen } from "@testing-library/react";

describe("Container", () => {
  it("should render with width: 100% by default", () => {
    renderWithTheme(
      <Container size="fluid">
        <h1>Container</h1>
      </Container>
    );

    expect(
      screen.getByRole("heading", { name: "Container" }).parentElement
    ).toHaveStyle({
      width: "100%"
    });
  });

  it("should render with max-width: 45rem when size is fixed", () => {
    const { container } = renderWithTheme(
      <Container size="fixed">
        <h1>Container</h1>
      </Container>
    );

    expect(
      screen.getByRole("heading", { name: "Container" }).parentElement
    ).toHaveStyle({
      "max-width": "45rem"
    });

    expect(container).toMatchSnapshot();
  });
});
