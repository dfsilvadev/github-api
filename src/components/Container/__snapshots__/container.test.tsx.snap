// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container should render with max-width: 45rem when size is fixed 1`] = `
.c0 {
  height: 100%;
  margin: 0 auto;
  padding: 2.5rem 1rem;
  max-width: 45rem;
}

<div>
  <div
    class="c0"
  >
    <h1>
      Container
    </h1>
  </div>
</div>
`;
